NewBatteryState.h

```diff
@@ -0,0 +1,39 @@
+#include "Battery.h"
+class NewBatteryState : public Battery
+{
+  public:
+    bool checkTempRange(float temperature)
+    {
+        if (temperature < 0 || temperature > 45)
```

Comment:
Similar range checking duplicate code can be reduced and refactored to a single method. for ex. IsValueInRange(...)